manager/controllers/utils.go

```diff
@@ -323,49 +323,92 @@ func (t *Tag) ApplyTag(tx *gorm.DB) *gorm.DB {
 	return tx.Where("h.tags @> ?::jsonb", query)
 }
 
-// Filter systems by tags,
-func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB, bool, error) {
-	if !enableCyndiTags {
-		return tx, false, nil
-	}
-	var applied bool
-
-	subq := database.Db.
-		Table("inventory.hosts h").
-		Select("h.id")
-
+func ParseTagsFilters(c *gin.Context) (map[string]FilterData, error) {
+	filters := Filters{}
 	tags := c.QueryArray("tags")
 	for _, t := range tags {
 		tag, err := ParseTag(t)
 		if err != nil {
 			LogAndRespBadRequest(c, err, err.Error())
-			return tx, false, err
+			return nil, err
+		}
+		key := *tag.Namespace + "/" + tag.Key
+		var value []string
+		if value = []string{}; tag.Value != nil {
+			value = strings.Split(*tag.Value, ",")
+		}
+		filters[key] = FilterData{
+			Operator: "eq",
+			Values:   value,
 		}
-		subq = tag.ApplyTag(subq)
-		applied = true
 	}
 
-	// Additional filters
 	filter := NestedQueryMap(c, "filter").Path("system_profile")
 	if filter != nil {
 		filter.Visit(func(path []string, val string) {
 			if len(path) == 1 && path[0] == "sap_system" {
-				subq = subq.Where("(h.system_profile ->> 'sap_system')::text = ?", val)
-				applied = true
+				filters["sap_system"] = FilterData{
+					Operator: "eq",
+					Values:   strings.Split(val, ","),
+				}
 			}
 			if len(path) >= 1 && path[0] == "sap_sids" {
 				val = fmt.Sprintf(`"%s"`, val)
-				subq = subq.Where("(h.system_profile ->> 'sap_sids')::jsonb @> ?::jsonb", val)
-				applied = true
+				var op string
+				if op = "eq"; len(path) > 1 {
+					op = path[1]
+				}
+				filters["sap_sids"] = FilterData{
+					Operator: op,
+					Values:   strings.Split(val, ","),
+				}
 			}
 		})
 	}
 
+	return filters, nil
+}
+
+// Filter systems by tags,
+func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB, bool, map[string]FilterData, error) {
```

Comment:
Good, but it would be better to have methods:
`ApplyTagsFilter(filterData map[string]FilterData, tx *gorm.DB, systemIDExpr string) (*gorm.DB, bool, error)`
- It reduces number of output arguments in `ApplyTagsFilter` method.

Reply:
... so we firstly call `ParseTagsFilters(c)` and then use the output to both filters application and adding to response metadata.